### PR TITLE
First add for minizip

### PR DESCRIPTION
**Pull request by ryanskeith**
Minizip 4.0.3

**Destination channel:**
main

### Links

- [3669](https://anaconda.atlassian.net/jira/software/c/projects/PKG/boards/222/backlog?selectedIssue=PKG-3659) 
- [Upstream repository](https://github.com/zlib-ng/minizip-ng)

### Explanation of changes:
- added gtest to avoid code from downloading from source and building
- added gtest to ignore_run_exports as it is a test dependency and not a run dependency

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -41,8 +44,7 @@ test:
     - test ! -f $PREFIX/lib/libminizip.a                # [unix]
 
 about:
-  home: https://github.com/zlib-ng/minizip-ng
-  # http://zlib.net/zlib_license.html
+  home: https://github.com/zlib-ng/minizicdp-ng
```
Comment (Jrice1317) [inline recipe/meta.yaml]:
```suggestion
  home: https://github.com/zlib-ng/minizip-ng
```

File: recipe/meta.yaml [hunk 2]
```diff
@@ -13,6 +13,8 @@ build:
   number: 0
   run_exports:
     - {{ pin_subpackage('minizip', max_pin='x') }}
+  ignore_run_exports:
+    - gtest
 
 requirements:
   build:
```
Comment (danpetry) [inline recipe/meta.yaml:20]: could probably run with ninja on unix too, probably only worth making the change if the build currently takes a long time though

**Review by danpetry — COMMENTED** [review]
A few comments for where I couldn't put comments in the code:
- probably better to explicitly state the [build options](https://github.com/zlib-ng/minizip-ng/blob/master/README.md#build-options)that are being used, for documentation purposes. I mean the ones that are default on and are already being used because the dependencies are there in the host environment
- you could build tests and unit tests for windows too
- do you know why the static library is included in the package for windows but not unix?
- also why CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS is on?

File: recipe/meta.yaml (hunk 2 above)
Comment (ryanskeith) [reply recipe/meta.yaml:20]: This seems pretty quick.

**Comment by ryanskeith** [comment]
> A few comments for where I couldn't put comments in the code:
> 
> * probably better to explicitly state the [build options](https://github.com/zlib-ng/minizip-ng/blob/master/README.md#build-options)that are being used, for documentation purposes. I mean the ones that are default on and are already being used because the dependencies are there in the host environment
> * you could build tests and unit tests for windows too
> * do you know why the static library is included in the package for windows but not unix?
> * also why CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS is on?

- That's a good idea on being explicit.  I will flesh those out.

- I will try adding tests for windows.
- I do not know why static libs are included in the package for windows and not unix.
- I am not sure why Windows Export all symbols was chosen as well.

**Review by skupr-anaconda — APPROVED** [review]
LGTM but you can use `ninja-base` instead of `ninja`

**Review by danpetry — APPROVED** [review]
>I do not know why static libs are included in the package for windows and not unix.

>I am not sure why Windows Export all symbols was chosen as well.

ok, I don't think it's a big deal, the package isn't used anywhere at the moment

**Comment by ryanskeith** [comment]
I found out why windows exports are the way they are found in this issue that is yet to be resolved: https://github.com/zlib-ng/minizip-ng/issues/475

**Pull request merged**